File: HVACCalculations.java
public class HVACCalculations {

    /**
     * Calculate heating power based on flow rates and temperature differences.
     * 
     * @param flowRate Flow rate in cubic meters per second (m³/s)
     * @param deltaT Temperature difference in Kelvin (K)
     * @param specificHeat Specific heat capacity of the fluid in Joules per kilogram Kelvin (J/(kg·K))
     * @return Heating power in Watts (W)
     */
    public static double calculateHeatingPower(double flowRate, double deltaT, double specificHeat) {
        return flowRate * deltaT * specificHeat;
    }

    /**
     * Estimate cooling power based on heat extraction requirements.
     * 
     * @param heatGain Heat gain in Watts (W)
     * @return Cooling power in Watts (W)
     */
    public static double estimateCoolingPower(double heatGain) {
        return heatGain;
    }

    /**
     * Calculate pressure drop in pipes using the Darcy-Weisbach equation.
     * 
     * @param length Length of the pipe in meters (m)
     * @param diameter Diameter of the pipe in meters (m)
     * @param flowVelocity Flow velocity in meters per second (m/s)
     * @param frictionFactor Darcy friction factor (dimensionless)
     * @param density Density of the fluid in kilograms per cubic meter (kg/m³)
     * @return Pressure drop in Pascals (Pa)
     */
    public static double calculatePressureDrop(double length, double diameter, double flowVelocity, double frictionFactor, double density) {
        return (frictionFactor * length * density * Math.pow(flowVelocity, 2)) / (2 * diameter);
    }
}

File: hvac_calculations.py
def heating_power(flow_rate, temp_diff):
    """
    Calculate heating power in watts.
    
    Parameters:
    flow_rate : float : flow rate in cubic meters per second
    temp_diff : float : temperature difference in degrees Celsius
    
    Returns:
    float : heating power in watts
    """
    specific_heat = 4184  # specific heat of water in J/(kg*K)
    density_water = 1000  # kg/m^3
    power = flow_rate * density_water * specific_heat * temp_diff
    return power


def cooling_power(flow_rate, temp_diff):
    """
    Estimate cooling power in watts.
    
    Parameters:
    flow_rate : float : flow rate in cubic meters per second
    temp_diff : float : temperature difference in degrees Celsius
    
    Returns:
    float : cooling power in watts
    """
    # Assume similar calculations as heating, but negative temp_diff for cooling
    return heating_power(flow_rate, -temp_diff)


def pressure_drop(length, diameter, flow_rate, viscosity):
    """
    Calculate pressure drop in pipes using the Darcy-Weisbach equation.
    
    Parameters:
    length : float : length of the pipe in meters
    diameter : float : diameter of the pipe in meters
    flow_rate : float : flow rate in cubic meters per second
    viscosity : float : dynamic viscosity of the fluid in Pa.s
    
    Returns:
    float : pressure drop in Pascals
    """
    # Calculate the Reynolds number
    area = 3.14159 * (diameter / 2) ** 2
    velocity = flow_rate / area
    density = 1000  # Assume density of water in kg/m^3
    reynolds_number = (density * velocity * diameter) / viscosity
    # Calculate friction factor using the Darcy-Weisbach equation
    if reynolds_number < 2000:
        # Laminar flow
        friction_factor = 64 / reynolds_number
    else:
        # Turbulent flow Poiseuille approximation
        friction_factor = 0.3164 / (reynolds_number ** 0.25)
    pressure_drop = friction_factor * (length / diameter) * (density * velocity ** 2) / 2
    return pressure_drop
